vllm/model_executor/models/qwen2.py

```diff
@@ -340,6 +350,11 @@ def load_weights(self,
                 model_name_or_path, cache_dir, load_format, revision):
             if "rotary_emb.inv_freq" in name:
                 continue
+            if self.config.tie_word_embeddings:
+                if "lm_head.weight" in name:
+                    continue
+                if name == "model.embed_tokens.weight":
+                    name = "lm_head_weight"
```

Comment:
A dumb question: Could you explain why we need this? IIUC, `lm_head.weight` will not be included in the checkpoint when `tie_word_embeddings=True`. Also, `model.embed_tokens.weight` should be loaded to `model.embed_tokens` rather than `lm_head_weight`.

Reply:
1. Current `Qwen1.5-0.5B-Chat` checkpoint on HuggingFace (incorrectly) includes a redundant `lm_head.weight`. (The checkpoint will be updated ASAP, but there are still users using old version checkpoint, so we manually skip `lm_head.weight` from checkpoint to avoid KeyError).
2. Line `self.lm_head_weight = self.model.embed_tokens.weight` in `__init__` method will remove the old name `model.embed_tokens.weight` from the module and re-register with new name `lm_head_weight`, so we need to change the name in `load_weights`.

Reply:
1. Thanks for the explanation!
2. If that's the case, can we directly use `self.model.embed_tokens.weight` instead of defining `lm_head_weight`?

Reply:
@WoosukKwon Hi, I have simplify the code with your suggestion, please kindly check it again.